OuterWildsRandomSpeedrun/SpawnPointMenu/SpawnPointSelectorManager.cs

```diff
@@ -0,0 +1,200 @@
+using OWML.Common;
+using SpawnPointSelector;
+using UnityEngine;
+using System;
+using System.Collections.Generic;
+using static UnityEngine.InputSystem.InputAction;
+
+namespace OuterWildsRandomSpeedrun
+{
+  public class SpawnPointSelectorManager : MonoBehaviour
+  {
+    public IModHelper ModHelper { get; set; }
+
+    public static SpawnPointSelectorManager Instance {
+      get => _instance;

```

Comment:
given that this is static, we should consider doing a true singleton and instantiating it here, or otherwise better handling the fact that instance might be null

Reply:
I had done this originally, but it caused problems. Specifically:
- From what I have found, Unity will destroy objects that aren't associated with a GameObject
- If I instantiate this in the getter, a second instance is created anyway when we call `AddComponent` in `OuterWildsRandomSpeedrun`, skipping the singleton. This led to some pretty weird behavior when leaving & re-entering the menu. That's the main reason this is being instantiated in `Awake()` instead. Let me know if you know of a better way though!

Reply:
Fixed over Discord discussion